<a href="https://colab.research.google.com/github/KacperKaszuba0608/Projects/blob/main/Exploring_eBay_Car_Sales_Data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Guided Project: Exploring eBay Car Sales Data

In this project we focused on exploring eBay Car Sales Data from Kaggle where the original dataset isn't availble now, but you can find it on [data.word](https://data.world/data-society/used-cars-data). Dataquest team sampled 50000 rows and change some values. They did it because we tried clean missing or understood values and then explore them. Below you have description of columns:

* `dateCrawled` - when this ad was first crawled. All field-values are taken from this date
* `name` - name of the car
* `seller` - private or dealer
* `offerType` - the tipe of listing
* `price` - the price on the ad to sell the car
* `abtest` - whether the listing is included in an A/B test
* `vehicleType`
* `yearOfRegistration` - at which year the car was first registered
* `gearbox` - the transmisson type
* `powerPS` - power of the car in PS
* `model` - the car model name
* `kilometer` - how many kilometers the car has driven
* `monthOfRegistration` - at which month the car was first registered
* `fuelType`
* `brand`
* `notRepairedDamage` - if the car has a damage which is not repaired yet
* `dateCreated` - the date for which the ad at ebay was created
* `nrOfPictures` - number of pictures in the ad
* `postalCode` - the postal code for the location of the vehicle
* `lastSeen` - when the crawler saw this ad last online

## Importing libraries and dataset

In [ ]:
import pandas as pd
import numpy as np

cars = pd.read_csv('https://query.data.world/s/b7vnijn6k5r4awstrt4xkouqwff3mi', encoding = 'Latin-1')

In [ ]:
cars.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371528 entries, 0 to 371527
Data columns (total 20 columns):
 #   Column               Non-Null Count   Dtype 
---  ------               --------------   ----- 
 0   dateCrawled          371528 non-null  object
 1   name                 371528 non-null  object
 2   seller               371528 non-null  object
 3   offerType            371528 non-null  object
 4   price                371528 non-null  int64 
 5   abtest               371528 non-null  object
 6   vehicleType          333659 non-null  object
 7   yearOfRegistration   371528 non-null  int64 
 8   gearbox              351319 non-null  object
 9   powerPS              371528 non-null  int64 
 10  model                351044 non-null  object
 11  kilometer            371528 non-null  int64 
 12  monthOfRegistration  371528 non-null  int64 
 13  fuelType             338142 non-null  object
 14  brand                371528 non-null  object
 15  notRepairedDamage    299468 non-nu

In [ ]:
cars.head()

dateCrawled                            name  seller offerType  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat   Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat   Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat   Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat   Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat   Angebot   

   price abtest vehicleType  yearOfRegistration    gearbox  powerPS  model  \
0    480   test         NaN                1993    manuell        0   golf   
1  18300   test       coupe                2011    manuell      190    NaN   
2   9800   test         suv                2004  automatik      163  grand   
3   1500   test  kleinwagen                2001    manuell       75   golf   
4   3600   test  kleinwagen                2008    manuell       69  fabia   

   kilometer  monthOfRegistration fuelType       brand notRepairedDamage  \
0     150000                    0   benzin  volkswagen               NaN   
1     125000                    5   diesel        audi                ja   
2     125000                    8   diesel        jeep               NaN   
3     150000                    6   benzin  volkswagen              nein   
4      90000                    7   diesel       skoda              nein   

           dateCreated  nrOfPictures  postalCode             lastSeen  
0  2016-03-24 00:00:00             0       70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00             0       66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00             0       90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00             0       91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00             0       60437  2016-04-06 10:17:21

As we see above the dataset has 50000 rows and 20 columns. Most of them are `object` type and only 5 columns have `int64` type. 5 columns have null values, so we had to clean it and some of column names are no in the same type. That's we changed them below.

## Cleaning Column Names

In [ ]:
cars.columns = ['crawled_date', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'pictures_num', 'postal_code',
       'last_seen']
cars.head()

ValueError: ignored

Now, column names are more simple and understood. We created list of new column names and asigned it to `cars.columns` function from pandas.

## Exploration and Cleaning

In [ ]:
cars.describe(include = 'all')

crawled_date         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-03-21 20:37:19  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   
mean                    NaN          NaN     NaN        NaN    NaN    NaN   
std                     NaN          NaN     NaN        NaN    NaN    NaN   
min                     NaN          NaN     NaN        NaN    NaN    NaN   
25%                     NaN          NaN     NaN        NaN    NaN    NaN   
50%                     NaN          NaN     NaN        NaN    NaN    NaN   
75%                     NaN          NaN     NaN        NaN    NaN    NaN   
max                     NaN          NaN     NaN        NaN    NaN    NaN   

       vehicle_type  registration_year  gearbox      power_ps  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  pictures_num   postal_code            last_seen  
count                 50000       50000.0  50000.000000                50000  
unique                   76           NaN           NaN                39481  
top     2016-04-03 00:00:00           NaN           NaN  2016-04-07 06:17:27  
freq                   1946           NaN           NaN                    8  
mean                    NaN           0.0  50813.627300                  NaN  
std                     NaN           0.0  25779.747957                  NaN  
min                     NaN           0.0   1067.000000                  NaN  
25%                     NaN           0.0  30451.000000                  NaN  
50%                     NaN           0.0  49577.000000                  NaN  
75%                     NaN           0.0  71540.000000                  NaN  
max                     NaN           0.0  99998.000000                  NaN

Description of columns tels us that 2 columns: `odometer` and `price` have combination of string and number. That's why we had to found a pattern and cleaned it. Column `registration_year` has a strange maximum value equal 9999. It's probably wrong value because 9999 year is a future. Column `power_ps` has unbelievable maximum value qual 177700 ps. None car has power like this one. Column `registration_month` has minimum value 0.00, month 0 doesn't exist so it's misstake too. We colud remove columns which every value were the same. It happened in columns: `seller`, `offer_type` and `pictures_num`. They told us nothing so we could remove it. What we did at first.

In [ ]:
cars = cars.drop(['seller', 'offer_type', 'pictures_num'], axis = 1)
cars.shape

(50000, 17)

In [ ]:
cars['price'] = cars['price'].str.replace('$', '').str.replace(',', '')
cars['price'] = cars['price'].astype(int)

In [ ]:
cars['odometer'] = cars['odometer'].str.replace('km', '').str.replace(',', '')
cars['odometer'] = cars['odometer'].astype(int)

In [ ]:
cars.rename({'odometer' : 'odometer_km', 'price' : 'price_dol'}, 
            inplace = True, axis = 1)

## Exploring and cleaning prices

In [ ]:
cars['price_dol'].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price_dol, dtype: float64

In [ ]:
cars['price_dol'].value_counts().sort_index(ascending = False).head(20)

99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
3890000     1
1300000     1
1234566     1
999999      2
999990      1
350000      1
345000      1
299000      1
295000      1
265000      1
259000      1
250000      1
220000      1
198000      1
197000      1
Name: price_dol, dtype: int64

In [ ]:
cars['price_dol'].value_counts().sort_index(ascending = True).head(20)

0     1421
1      156
2        3
3        1
5        2
8        1
9        1
10       7
11       2
12       3
13       2
14       1
15       2
17       3
18       1
20       4
25       5
29       1
30       7
35       1
Name: price_dol, dtype: int64

As we see above we have cars which cost 0\\$ and cars which cost 999990$ and more. That's why we chose cars' prices between 1\\$ and 350001\\$. We did it below.

In [ ]:
cars = cars[cars['price_dol'].between(1, 350001)]
cars['price_dol'].describe()

count     48565.000000
mean       5888.935591
std        9059.854754
min           1.000000
25%        1200.000000
50%        3000.000000
75%        7490.000000
max      350000.000000
Name: price_dol, dtype: float64

## Exploring and cleaning odometer_km

In [ ]:
cars['odometer_km'].describe()

count     48565.000000
mean     125770.101925
std       39788.636804
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

As we see odometer column is rounded and we decided that nothing in it is mistake and we didn't have to change anything.

## Exploring and cleaning registration_year and month

In [ ]:
cars['registration_year'].describe()

count    48565.000000
mean      2004.755421
std         88.643887
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

Above, maximum years is 9999 and it's imposible. Minimum value is 1000 and it historically imposible too. That's we had to find registration year between 1886 (date of creation first car) and now (2022).

In [ ]:
cars = cars[cars['registration_year'].between(1886, 2022)]

In [ ]:
cars['registration_year'].describe()

count    48545.000000
mean      2003.461510
std          7.566196
min       1910.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       2019.000000
Name: registration_year, dtype: float64

In [ ]:
cars['registration_month'].describe()

count    48545.000000
mean         5.783603
std          3.684938
min          0.000000
25%          3.000000
50%          6.000000
75%          9.000000
max         12.000000
Name: registration_month, dtype: float64

In [ ]:
cars.loc[cars['registration_month'] == 0, 'registration_month'].value_counts()

0    4468
Name: registration_month, dtype: int64

`registration_month` column has one mistake. It has values equal 0. All know that months numbers are 1 to 12, so we had to drop rows with 0. There were 4468 of these values.

In [ ]:
cars = cars[cars['registration_month'].between(1,12)]

In [ ]:
cars['registration_month'].describe()

count    44077.000000
mean         6.369875
std          3.349729
min          1.000000
25%          3.000000
50%          6.000000
75%          9.000000
max         12.000000
Name: registration_month, dtype: float64

In [ ]:
(cars['registration_year'].value_counts()
                          .sort_values(ascending = False)
                          .head(15))

2005    2698
1999    2614
2000    2594
2004    2587
2006    2576
2003    2538
2001    2416
2002    2323
2007    2194
2008    2138
1998    2077
2009    2034
1997    1703
2011    1595
2010    1562
Name: registration_year, dtype: int64

The most cars registrated 20 years ago

## Exploring and cleaning power_ps

In [ ]:
cars['power_ps'].describe()

count    44077.000000
mean       121.987045
std        207.906287
min          0.000000
25%         75.000000
50%        110.000000
75%        150.000000
max      17700.000000
Name: power_ps, dtype: float64

The fastest car in the world has 1600ps, so this to upper limit of ps value, that why every value over it has to be remove. The car can't have power qual 0, so we had to check it too.

In [ ]:
cars['power_ps'].value_counts().sort_index().head(20)

0     3266
1        5
2        1
3        2
4        3
5       10
6        3
8        2
10       2
11       4
14       1
15       3
18       4
19       1
20       2
21       1
23       3
24       1
25       1
26      23
Name: power_ps, dtype: int64

In [ ]:
cars.loc[cars['power_ps'] > 1600, 'power_ps'].value_counts().sum()

28

In [ ]:
cars = cars[cars['power_ps'].between(1, 1601)]

In [ ]:
cars['power_ps'].describe()

count    40783.000000
mean       127.790182
std         67.077762
min          1.000000
25%         80.000000
50%        116.000000
75%        150.000000
max       1405.000000
Name: power_ps, dtype: float64

In [ ]:
cars['power_ps'].value_counts().sort_values(ascending = False).head(10)

75     2785
60     1937
150    1833
140    1790
101    1586
116    1537
90     1535
170    1404
105    1339
136     868
Name: power_ps, dtype: int64

The most cars have power between 60 and 150 ps.

## Exploring the date columns

In [ ]:
cars[['crawled_date','ad_created','last_seen']][0:5]

crawled_date           ad_created            last_seen
0  2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
1  2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08
2  2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37
3  2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
5  2016-03-21 13:47:45  2016-03-21 00:00:00  2016-04-06 09:45:21

In [ ]:
(cars['crawled_date'].str[:10]
                     .value_counts(normalize = True, dropna = False)
                     .sort_index())

2016-03-05    0.025403
2016-03-06    0.014295
2016-03-07    0.036241
2016-03-08    0.032906
2016-03-09    0.032317
2016-03-10    0.032366
2016-03-11    0.032489
2016-03-12    0.037270
2016-03-13    0.015570
2016-03-14    0.036854
2016-03-15    0.034009
2016-03-16    0.029301
2016-03-17    0.031214
2016-03-18    0.012922
2016-03-19    0.034451
2016-03-20    0.038398
2016-03-21    0.037270
2016-03-22    0.032881
2016-03-23    0.031631
2016-03-24    0.029179
2016-03-25    0.031459
2016-03-26    0.032587
2016-03-27    0.030920
2016-03-28    0.035309
2016-03-29    0.033372
2016-03-30    0.033298
2016-03-31    0.032170
2016-04-01    0.034279
2016-04-02    0.035603
2016-04-03    0.039158
2016-04-04    0.036976
2016-04-05    0.013290
2016-04-06    0.003188
2016-04-07    0.001422
Name: crawled_date, dtype: float64

In [ ]:
(cars['last_seen'].str[:10]
                     .value_counts(normalize = True, dropna = False)
                     .sort_index())

2016-03-05    0.001030
2016-03-06    0.003899
2016-03-07    0.005051
2016-03-08    0.006792
2016-03-09    0.009220
2016-03-10    0.010274
2016-03-11    0.011868
2016-03-12    0.023466
2016-03-13    0.008459
2016-03-14    0.012407
2016-03-15    0.015497
2016-03-16    0.015913
2016-03-17    0.027119
2016-03-18    0.007233
2016-03-19    0.015644
2016-03-20    0.020254
2016-03-21    0.019959
2016-03-22    0.020817
2016-03-23    0.018292
2016-03-24    0.019273
2016-03-25    0.018856
2016-03-26    0.016330
2016-03-27    0.014884
2016-03-28    0.020474
2016-03-29    0.021921
2016-03-30    0.024667
2016-03-31    0.023294
2016-04-01    0.023269
2016-04-02    0.025231
2016-04-03    0.024839
2016-04-04    0.023809
2016-04-05    0.128264
2016-04-06    0.227055
2016-04-07    0.134639
Name: last_seen, dtype: float64

As we see above, columns `crawled_date` and `last_seen` are percentages similar. The ads were crawled day by day for two months (March and April). Three last rows in `last_seen` column are diferent and disturb the rest of distribution. This values are higher than earlier but it's not because of sale many car, but probably bacause ending the crawled.

In [ ]:
(cars['ad_created'].str[:10]
                     .value_counts(normalize = True, dropna = False)
                     .sort_index())

2015-08-10    0.000025
2015-09-09    0.000025
2015-11-10    0.000025
2015-12-05    0.000025
2015-12-30    0.000025
                ...   
2016-04-03    0.039355
2016-04-04    0.037369
2016-04-05    0.011990
2016-04-06    0.003286
2016-04-07    0.001251
Name: ad_created, Length: 74, dtype: float64

The most ads were created in two months (March and April) and that months tally with `crawled_date` and `last_seen` columns. But in `ad_created` column we saw records which had ads created 9 month earlier than first record in `crawled_date` column.

## Exploring Price by brand

We chose top 20 brands and explore mean price of car.

In [ ]:
def means_by_brand(dataset, brand_column, mean_column):
    '''
    dataset - variable with data (dataframe)
    brand_column - name of column with brands (string)
    mean_column - name of column witch we want calcuate mean (string)
    '''
    mean_by_brands = {}
    
    brands = dataset[brand_column].value_counts(normalize = True).head(20).index
    
    for brand in brands:
        one_brand = dataset[dataset[brand_column] == brand]
        mean_value = one_brand[mean_column].mean()
        mean_by_brands[brand] = int(mean_value)
        
    return mean_by_brands

In [ ]:
mean_price = means_by_brand(cars, 'brand', 'price_dol')
mean_price

{'volkswagen': 5887,
 'bmw': 8793,
 'opel': 3272,
 'mercedes_benz': 9083,
 'audi': 9958,
 'ford': 4062,
 'renault': 2705,
 'peugeot': 3318,
 'fiat': 3035,
 'seat': 4722,
 'skoda': 6748,
 'nissan': 5201,
 'mazda': 4513,
 'smart': 3775,
 'citroen': 3973,
 'toyota': 5366,
 'hyundai': 5753,
 'mini': 10719,
 'volvo': 5191,
 'honda': 4410}

Top 5 most expensive cars are:
1. mini
2. audi
3. mercedes_benz
4. bmw
5. skoda

So we can conclude that:
* Audi, mercedes and bwm are expensive and luxury cars. 
* Opel and ford are chipper cars but still good.
* Skoda and volkswagen are the best option for both fiancial status. 

We deduct some brands in conclusion because some of them has low value of percentage of the total values.

## Storing Aggregate Data in a DataFrame

Below we created new data frame which store aggraagate data about mean price and mean mileage for top 20 brands.

In [ ]:
mileage = means_by_brand(cars, 'brand', 'odometer_km')

In [ ]:
brand_info = pd.DataFrame(pd.Series(mean_price), 
                          columns=['mean_price'])
brand_info['mileage'] = pd.Series(mileage)

In [ ]:
brand_info

mean_price  mileage
volkswagen           5887   127832
bmw                  8793   132327
opel                 3272   128519
mercedes_benz        9083   130502
audi                 9958   128279
ford                 4062   123761
renault              2705   126797
peugeot              3318   125756
fiat                 3035   116380
seat                 4722   120427
skoda                6748   109505
nissan               5201   116473
mazda                4513   123180
smart                3775    98273
citroen              3973   118419
toyota               5366   115018
hyundai              5753   104539
mini                10719    88524
volvo                5191   138363
honda                4410   122286

As we see above, all cars have similar mileage what confirm our earlier conclusion. And at the end of this project we can say that opel and ford are cheap cars with same mileage like 'luxury' cars (mean bmw, audi and mercedes_benz). Namely skoda and volkswagen are for both clients with high standrads and low. We can ad that skoda has lower mileage than volkswagen but volkswagen is cheaper. So, something for something.

## Translating german words to english eqivalent

In [ ]:
cars[['vehicle_type', 'gearbox', 
      'model', 'fuel_type', 'unrepaired_damage']].head(10)

vehicle_type    gearbox    model fuel_type unrepaired_damage
0           bus    manuell   andere       lpg              nein
1     limousine  automatik      7er    benzin              nein
2     limousine    manuell     golf    benzin              nein
3    kleinwagen  automatik   fortwo    benzin              nein
5           bus  automatik  voyager    diesel               NaN
6     limousine    manuell     golf    benzin               NaN
7     limousine    manuell     golf    diesel              nein
9           bus    manuell   megane    benzin              nein
10          NaN    manuell      NaN    benzin              nein
12   kleinwagen  automatik   fortwo    benzin              nein

As we see, columns above shows strings in german. Our chalange were translating this words to english eqivalent. We did it below.

### Vehicle_type

In this column we had to change only two german words:
* kleinwagen to small cars
* andere to other

In [ ]:
cars['vehicle_type'].unique()

array(['bus', 'limousine', 'kleinwagen', nan, 'coupe', 'kombi', 'suv',
       'cabrio', 'andere'], dtype=object)

In [ ]:
cars.loc[cars['vehicle_type'] == 'kleinwagen', 'vehicle_type'] = 'small car'
cars.loc[cars['vehicle_type'] == 'andere', 'vehicle_type'] = 'other'

In [ ]:
cars['vehicle_type'].unique()

array(['bus', 'limousine', 'small car', nan, 'coupe', 'kombi', 'suv',
       'cabrio', 'other'], dtype=object)

### Gearbox

Here, we had to change all words to english:
* manuell to manual
* automatik to automatic

In [ ]:
cars['gearbox'].unique()

array(['manuell', 'automatik', nan], dtype=object)

In [ ]:
cars.loc[cars['gearbox'] == 'manuell', 'gearbox'] = 'manual'
cars.loc[cars['gearbox'] == 'automatik', 'gearbox'] = 'automatic'

In [ ]:
cars['gearbox'].unique()

array(['manual', 'automatic', nan], dtype=object)

### Car model

In this column german words are "andere" which we had to change to "other". Of course dataset has more german words but changing them will be difficult and very monotonous.

In [ ]:
cars.loc[cars['model'] == 'andere', 'model'] = 'other'

### Fuel type

Here, we had the same situation like above and we just change 'andere' word.

In [ ]:
cars['fuel_type'].unique()

array(['lpg', 'benzin', 'diesel', nan, 'cng', 'hybrid', 'elektro',
       'andere'], dtype=object)

In [ ]:
cars.loc[cars['fuel_type'] == 'andere', 'fuel_type'] = 'other'

In [ ]:
cars['fuel_type'].unique()

array(['lpg', 'benzin', 'diesel', nan, 'cng', 'hybrid', 'elektro',
       'other'], dtype=object)

### Unrepaired damage

This column has two german words ("nein", "ja"). We changed it to english "no" and "yes".

In [ ]:
cars['unrepaired_damage'].unique()

array(['nein', nan, 'ja'], dtype=object)

In [ ]:
cars.loc[cars['unrepaired_damage'] == 'ja', 'unrepaired_damage'] = 'yes'
cars.loc[cars['unrepaired_damage'] == 'nein', 'unrepaired_damage'] = 'no'

In [ ]:
cars['unrepaired_damage'].unique()

array(['no', nan, 'yes'], dtype=object)

## How much cheaper are cars with damage than their non-damaged counterparts?

In [ ]:
damaged_cars_price = round(cars.loc[cars['unrepaired_damage'] == 'yes',
                              'price_dol'].mean(), 4)
non_damaged_cars_price = round(cars.loc[cars['unrepaired_damage'] == 'no',
                              'price_dol'].mean(), 4)

print(f'Damged cars cost {damaged_cars_price}$\nNon-damaged cars cost {non_damaged_cars_price}$')

Damged cars cost 2461.3647$
Non-damaged cars cost 7367.1156$


Based on average of prices, concluding is non-damaged cars are expensive 3 times more